lib/form/SettingsPermissionsForm.class.php

```diff
@@ -26,85 +26,69 @@
-      'disallow_thumb'         => new sfWidgetFormSelect(array('choices' => $choices))
-    ));
-
-    $form->widgetSchema->setLabels(array(
-      'allow_master'           => $i18n->__('View / download master representation'),
-      'allow_reference'        => $i18n->__('View / download reference representation'),
-      'allow_thumb'            => $i18n->__('View / download thumbnail representation'),
-      'conditional_master'     => $i18n->__('View / download master representation'),
-      'conditional_reference'  => $i18n->__('View / download reference representation'),
-      'conditional_thumb'      => $i18n->__('View / download thumbnail representation'),
-      'disallow_master'        => $i18n->__('View / download master representation'),
-      'disallow_reference'     => $i18n->__('View / download reference representation'),
-      'disallow_thumb'         => $i18n->__('View / download thumbnail representation')
-    ));
+    $choices = array();
+    foreach (QubitTaxonomy::getTermsById(QubitTaxonomy::RIGHT_ACT_ID) as $item)
+    {
+      $choices[$item->slug] = $item->__toString();
+    }
+    $this->setWidget('granted_right', new sfWidgetFormSelect(array('choices' => $choices)));
+    $this->setDefault('granted_right', $premisAccessRight->value);
+    $this->setValidator('granted_right', new sfValidatorChoice(array('choices' => array_keys($choices))));
 
-    $form->setDefaults(unserialize($this->premisAccessRightValues->value));
+    //
+    // PREMIS permissionss
+    //
 
-    return $form;
+    $this->embedForm('permissions', $this->getPermissionsForm());
   }
 
-  public function configure()
+  protected function getPermissionsForm()
   {
-    $this->premisAccessRight = QubitSetting::getByName('premisAccessRight');
-    $this->premisAccessRightValues = QubitSetting::getByName('premisAccessRightValues');
-
-    if (null === $this->premisAccessRightValues)
+    $premisAccessRightValues = QubitSetting::getByName('premisAccessRightValues');
+    if (null === $premisAccessRightValues)
     {
-      $this->premisAccessRightValues = QubitSetting::createNewSetting(
-        'premisAccessRightValues', serialize(QubitSetting::$premisAccessRightValueDefaults), array('deleteable' => false)
-      );
-
-      $this->premisAccessRightValues->save();
+      throw new sfException('Setting premisAccessRightValues cannot be found');
     }
 
-    if (null === $this->premisAccessRight)
+    $premisAccessRightValues = unserialize($premisAccessRightValues->value);
+    $defaults = QubitSetting::$premisAccessRightValueDefaults;
+
+    $form = new sfForm;
+    $form->getValidatorSchema()->setOption('allow_extra_fields', true);
+
+    foreach (QubitTaxonomy::getTermsById(QubitTaxonomy::RIGHT_BASIS_ID) as $item)
     {
-      $this->premisAccessRight = QubitSetting::createNewSetting('premisAccessRight', 'delete', array('deleteable' => false));
-      $this->premisAccessRight->save();
+      $f = new sfForm;
```

Comment:
I think longer variable names make things clearer, but it's a matter of taste as they also add more text which could be considered visual clutter.